packages/ember-auto-import/ts/analyzer.ts

```diff
@@ -162,11 +162,68 @@ export default class Analyzer extends Plugin {
           // it's a syntax error to have anything other than exactly one
           // argument, so we can just assume this exists
           let argument = path.node.arguments[0];
-          if (argument.type !== 'StringLiteral') {
+
+          // Only a limited subset dynamic import syntax is allowed by v2
+          // addon package format. It must be either a string literal or a
+          // template string.
+          if (argument.type !== 'StringLiteral' && argument.type !== 'TemplateLiteral') {
             throw new Error(
-              'ember-auto-import only supports dynamic import() with a string literal argument.'
+              'ember-auto-import only supports dynamic import() that are ' +
+              'included in the supported subset of dynamic import syntax ' +
+              'of v2 addon format. Only string or template literal arguments ' +
+              `are allowed by that specification but ${argument.type} is used.`
             );
           }
+
+          // Only template strings that begins with a static prefix which
+          // matches a given list of cases are allowed.
+          if (
+            argument.type === 'TemplateLiteral'
+          ) {
+            // A template literal always starts with a static TemplateElement.
+            // For a template literal like `${foo}` that TemplateElement is
+            // an empty string.
+            let prefix =  argument.quasis[0].value.raw;
+
+            if (
+              // The static prefix may be an absolute URL either with a given
+              // protocol (`http://` or `https://`) or protocol-relative (`//`).
+              /(\w+:)?\/\//.test(prefix)
+            ) {
+              // In this case, Embroider will leave the import() alone. The
+              // browser's implementation of import() is used. The contents of
+              // the URL is beyond the scope of the Embroider build.
+              return;
+            } else if (
+              // The static prefix is a NPM package name or a relative path.
+              (
+                // This matches a namespaced NPM package name.
+                prefix.startsWith('@') && (prefix.match(/\//) || []).length >= 2
```

Comment:
I don't think it does. The length of result of `match` will never be > 1 because it doesn't capture anything. How about this instead:
```suggestion
                /^@[^/]+\/[^/]+$/.test(prefix)
```

Reply:
You are right. It was missing the global modifier:

```js
let prefix = '@foo/ab/';
prefix.match(/\/).length // 1
prefix.match(/\//g).length // 2
```

```suggestion
                prefix.startsWith('@') && (prefix.match(/\//g) || []).length >= 2
```

I decided for this implementation cause it's following the RFC closely:

> - the static prefix starts with @ and contains at least two / 
>   - this detects a namespaced NPM package name. If the package can't be resolved or it contains zero files that match your pattern, we emit a static build error.

I think your regular expression is not testing if the string contains a closing slash at the end. The prefix must not only be a valid NPM package name. It must also include a slash that marks the end of the package name. Otherwise the pattern would allow invalid patterns like this:

```js
let invalid = `@foo/bar${dynamic}`;
```

It should also allow additional characters as something like this is fully valid:

```js
let valid = `@foo/bar/baz/${dynamic}`;
```

Additionally it seems to miss some escaping of slashes. I think a working solution would look like this:

```suggestion
                /^@[^\/]+\/[^\/]+\//.test(prefix)
```

Not sure which one to prefer.

Reply:
Applied the fix to my preferred solution so that this one could be merged without any additional changes. Let me know if you still prefer the regular expression one. Open to refactor in that case.